hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -314,6 +314,65 @@ collectComTx networkId scriptRegistry vk headId headParameters (headInput, initi
   commitRedeemer =
     toScriptData $ Commit.redeemer Commit.ViaCollectCom
 
+-- | Construct a _decrement_ transaction which takes as input some 'UTxO' present
+-- in the L2 ledger state and makes it available on L1.
+decrementTx ::
+  NetworkId ->
+  -- | Published Hydra scripts to reference.
+  ScriptRegistry ->
+  -- | Party who's authorizing this transaction
+  VerificationKey PaymentKey ->
+  -- | Head identifier
+  HeadId ->
+  -- | Parameters of the head.
+  HeadParameters ->
+  -- | Everything needed to spend the Head state-machine output.
+  (TxIn, TxOut CtxUTxO) ->
+  -- | Decrement UTxO.
+  UTxO ->
+  -- | Confirmed Snapshot
+  Snapshot Tx ->
+  MultiSignature (Snapshot Tx) ->
+  Tx
+decrementTx networkId scriptRegistry vk headId headParameters (headInput, headOutput) utxoToDecrement snapshot signatures =
+  unsafeBuildTransaction $
+    emptyTxBody
+      & addInputs [(headInput, headWitness)]
+      & addReferenceInputs [headScriptRef]
+      & addOutputs (headOutput' : map toTxContext (toList utxoToDecrement))
+      & addExtraRequiredSigners [verificationKeyHash vk]
+ where
+  headRedeemer = toScriptData $ Head.Decrement (toPlutusSignatures signatures)
+  utxoHash = toBuiltin $ hashUTxO @Tx utxo
+
+  HeadParameters{parties, contestationPeriod} = headParameters
+
+  headOutput' =
+    TxOut
+      (mkScriptAddress @PlutusScriptV2 networkId headScript)
+      (txOutValue headOutput)
+      headDatumAfter
+      ReferenceScriptNone
```

Comment:
Should: avoid requiring a `networkId`, by just using `headOutput` and `modifyTxOutDatum` on it (like in the close)